app/src/main/java/com/android/shelfLife/model/recipe/RecipeGeneratorOpenAIRepository.kt

```diff
@@ -32,193 +27,186 @@ import kotlinx.serialization.json.putJsonArray
-                  ChatMessage(
-                      role = ChatRole.User,
-                      content =
-                          (userPrompt + USER_PROMPT_USE_TOOL_CALL) // User prompt with food items
-                      ))
-          tools {
-            function(
-                name = "_createRecipeFunction",
-                description =
-                    "Generate a recipe with a list of ingredients, step by step instructions, servings, and cooking time in seconds",
-                parameters = params)
+            putJsonArray("required") {
+              add("ingredients")
+              add("servings")
+              add("time")
+              add("instructions")
+            }
           }
-          toolChoice = ToolChoice.Auto // Automatically selects the tool
-        }
 
-        // Send the request to OpenAI and retrieve the tool call response
-        val response = openai.chatCompletion(request)
-        val message = response.choices.firstOrNull()?.message
-        message?.toolCalls?.firstOrNull()?.let { toolCall ->
-          require(toolCall is ToolCall.Function) { "Tool call is not a function" }
-          val toolResponse = toolCall.execute() as Map<String, Any>
-          // Construct the final Recipe object from the tool response
-          val generatedRecipe =
-              Recipe(
-                  uid = "0", // Placeholder UID
-                  name = recipePrompt.name,
-                  instructions = toolResponse["instructions"] as List<String>,
-                  servings = (toolResponse["servings"] as Int).toFloat(),
-                  time = (toolResponse["time"] as Long).minutes,
-                  ingredients =
-                      (toolResponse["ingredients"] as List<Map<String, Any>>).map { ingredient ->
-                        Ingredient(
-                            name = ingredient["name"] as String, // Access the map value for "name"
-                            quantity =
-                                Quantity(
-                                    amount =
-                                        ingredient["quantity"]
-                                            as Double, // Access the map value for "quantity"
-                                    unit =
-                                        ingredient["unit"]
-                                            as FoodUnit // Access the map value for "unit"
-                                    ),
-                            macros = NutritionFacts() // Default macros
-                            )
-                      },
-                  recipeType = recipePrompt.recipeType)
-          onSuccess(generatedRecipe) // Return the generated recipe
-        } ?: onFailure(Exception("No tool call generated"))
-      } catch (e: Exception) {
-        onFailure(e)
+      // Create the chat completion request with the system and user prompts
+      val request = chatCompletionRequest {
+        model = ModelId("gpt-4o-mini")
```

Comment:
Doesnt this have a limit? Why cant we use the regular chatgpt version (the one that's free)?

Reply:
Its cheaper XD